### PR TITLE
Use client-go instead of spawning kubectl, react on errors, postprocess output

### PR DESCRIPTION
**Pull request by pothos**
Forking kubectl processes with the same stdout was an
easy way to run processes on the pod.
Now client-go is used for that with the advantage to
run in go routines that are terminated at program exit
and don't write to stdout any further when the shell
prompt is already there.

The install subcommand now propagates the exit code.

The bcc subcommands post process the output of each node. This allows to detect multiple first lines and annotate to which node a line belongs. The subcommand now fails if one of the node processes fails.

### CODE REVIEW

File: cmd/inspektor-gadget/cmd/health.go [hunk 1]
```diff
@@ -54,13 +61,13 @@ func runHealth(cmd *cobra.Command, args []string) {
...
 	for _, node := range nodes.Items {
-		line := fmt.Sprintf("%s\t%s\t", node.Name, execPodQuick(client, node.Name, "/bin/gadget-node-health-check.sh"))
+		line := fmt.Sprintf("%s\t%s\t", node.Name, execPod(client, node.Name, "/bin/gadget-node-health-check.sh", true))
 		fmt.Fprintln(w, line)
 	}
 	w.Flush()
 }
 
-func execPodQuick(client *kubernetes.Clientset, node string, podCmd string) string {
+func execPod(client *kubernetes.Clientset, node string, podCmd string, capture bool) string {
```
Comment (alban) [inline cmd/inspektor-gadget/cmd/health.go]: could you move all of the helper functions that are not related to the `inspektor-gadget health` subcommand to another file than `health.go`? It's my fault, I put the helper functions there in the first place, but maybe that's a good opportunity to clean up this.

File: cmd/inspektor-gadget/cmd/health.go [hunk 2]
```diff
@@ -77,57 +84,67 @@ func execPodQuick(client *kubernetes.Clientset, node string, podCmd string) stri
...
+		err = exec.Stream(remotecommand.StreamOptions{
+			Stdin:  nil,
+			Stdout: &stdout,
+			Stderr: &stderr,
+			Tty:    false,
+		})
+		if err != nil {
+			return fmt.Sprintf("%s", err) + stdout.String() + stderr.String()
+		}
+		return stdout.String() + stderr.String()
```
Comment (alban) [inline cmd/inspektor-gadget/cmd/health.go]: I think the prototype of the function `execPod()` could be improved by returning a Golang `error` as well. So we would propagate the error in `install.go` and the command `inspektor-gadget install` could return a proper exit value.

File: cmd/inspektor-gadget/cmd/health.go (hunk 1 above)
Comment (pothos) [reply cmd/inspektor-gadget/cmd/health.go]: done

File: cmd/inspektor-gadget/cmd/health.go (hunk 2 above)
Comment (pothos) [reply cmd/inspektor-gadget/cmd/health.go]: Ok, I've prepared everything for stdout postprocessing through a custom `io.Writer` that, e.g., can read the first lines to detect failures and then prepend node identifiers (But that should be in another commit/PR).

File: cmd/inspektor-gadget/cmd/bcck8s.go [hunk 3]
```diff
@@ -81,6 +85,41 @@ func init() {
...
+	firstLine bool
+	firstLinePrinted *uint64
+	failure chan string
+}
+
+func (post postProcess) Write(p []byte) (n int, err error) {
+	prefix := "[" + post.nodeShort + "] "
+	asStr := ""
+        if len(p) != 0 {
+		asStr = string(p)
```
Comment (alban) [inline cmd/inspektor-gadget/cmd/bcck8s.go]: We don't seem to need to test len(p) and we can just write `asStr := string(p)`:
https://play.golang.org/p/3xsKw7KnXgE

Comment (pothos) [reply cmd/inspektor-gadget/cmd/bcck8s.go]: Thanks, yes, changed.

**Comment by alban** [comment]
Thanks!
